### PR TITLE
feat: post option follow

### PR DESCRIPTION
**Pull request by capJavert**
## Changes

- add option to menu
- mutations/invalidations of query cache work out of the box

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

<!--
Copy and paste the below line outside the HTML comment tags to link this PR to the ticket in Jira

AS-569

### CODE REVIEW

File: packages/shared/src/components/PostOptionsMenu.tsx [hunk 1]
```diff
@@ -372,6 +379,34 @@ export default function PostOptionsMenu({
     }
   }
 
+  if (isLoggedIn && post?.author) {
+    const authorName = post.author.name || post.author.username;
+    const isFollowingUser = [
+      ContentPreferenceStatus.Follow,
+      ContentPreferenceStatus.Subscribed,
```
Comment (rebelchris) [inline packages/shared/src/components/PostOptionsMenu.tsx:386]: Nothing urgent, but maybe we need to consolidate this "fn" at some point.
Pretty sure we are going to need it elsewhere.

File: packages/shared/src/components/PostOptionsMenu.tsx [hunk 2]
```diff
@@ -372,6 +379,34 @@ export default function PostOptionsMenu({
...
+          follow({
+            id: post.author.id,
+            entity: ContentPreferenceType.User,
+            entityName: authorName,
+          });
+        } else {
+          unfollow({
+            id: post.author.id,
+            entity: ContentPreferenceType.User,
+            entityName: authorName,
```
Comment (rebelchris) [inline packages/shared/src/components/PostOptionsMenu.tsx:403]: Also I did `@name` (probably should remove the @) but maybe we can have some rules around it internally?

**Review by rebelchris — APPROVED** [review]
This part looks good :) 
How come the invalidation worked here already?

File: packages/shared/src/components/PostOptionsMenu.tsx (hunk 1 above)
Comment (capJavert) [reply packages/shared/src/components/PostOptionsMenu.tsx]: Added util 6f1d155476649456dea14214d4298990d12160d9

File: packages/shared/src/components/PostOptionsMenu.tsx (hunk 2 above)
Comment (capJavert) [reply packages/shared/src/components/PostOptionsMenu.tsx:398]: yep dc3b29712bdfd86cf5408cfeb6bcada7606e0630

**Pull request merged**